db/models.py

```diff
@@ -50,3 +60,64 @@
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        to=User, on_delete=models.CASCADE, related_name="orders"
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return self.created_at.strftime("%Y-%m-%d %H:%M:%S")
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(
+        MovieSession, on_delete=models.CASCADE, related_name="tickets"
+    )
+    order = models.ForeignKey(
+        Order, on_delete=models.CASCADE, related_name="tickets"
+    )
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="unique_ticket"
+            )
+        ]
+
+    def clean(self) -> None:
+        cinema_hall = self.movie_session.cinema_hall
+        if self.row < 1 or self.row > cinema_hall.rows:
+            raise ValidationError(
+                {
+                    "row": (
+                        f"row number must be in available range: ("
+                        f"1, rows): (1, "
+                        f"{cinema_hall.rows})"
+                    )
+                }
+            )
+        if self.seat < 1 or self.seat > cinema_hall.seats_in_row:
+            raise ValidationError(
+                {
+                    "seat": (
+                        f"seat number must be in available range: "
+                        f"(1, seats_in_row): (1, {cinema_hall.seats_in_row})"
+                    )
+                }
+            )
+
+    def save(self, *args, **kwargs) -> None:
+        self.full_clean()
```

Comment:
The `full_clean` method is called before saving a `Ticket` instance to ensure data validation. This is a good practice to maintain data integrity.